docker-wrappers/SPRAS/snakemake_long.py

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python3
+
+"""
+A wrapper script that allows long-term Snakemake workflows to run on HTCondor. This works
+by submitting a local universe job responsible for overseeing the terminal session that
+runs the actual snakemake executable.
+"""
+
+import argparse
+import os
+import pathlib
+import subprocess
+import sys
+import time
+
+import htcondor
+
+"""
+Parse various arguments for the script. Note that this script has two "modes" of operation which
+need different arguments. The "top" mode is for submitting the HTCondor wrapper, and the "long" mode
+is for running the Snakemake command itself.
+"""
+def parseArgs(isLocal=False):
+    parser = argparse.ArgumentParser(description="A tool for long-running Snakemake jobs with HTCondor.")
+    # We add a special command to trigger allowing this script to execute the long-running Snakemake command.
+    if isLocal:
+        parser.add_argument("command", help="Helper command to run", choices=["long"])
+    parser.add_argument("--snakefile", help="The Snakefile to run.", required=False)
+    parser.add_argument("--profile", help="A path to a directory containing the desired Snakemake profile.", required=True)
+    # I'd love to change this to "logdir", but using the same name as Snakemake for consistency of feeling between this script
+    # and Snakemake proper.
+    parser.add_argument("--htcondor-jobdir", help="The directory Snakemake will write logs to.", required=False)
+    return parser.parse_args()
+
+"""
+Given a Snakefile, profile, and HTCondor job directory, submit a local universe job that runs
+Snakemake from the context of the submission directory.
+"""
+def submitLocal(snakefile, profile, htcondor_jobdir):
+    # Get the location of this script, which also serves as the executable for the condor job.
+    script_location = pathlib.Path(__file__).resolve()
+
+    submit_description = htcondor.Submit({
+        "executable":              script_location,
+        "arguments":               f"long --snakefile {snakefile} --profile {profile} --htcondor-jobdir {htcondor_jobdir}",
```

Comment:
Does `long` here imply there is a max runtime on the local universe job?

Reply:
Nope, it's my way of using the same executable to serve two purposes -- `snakemake_long.py` is first run by the user with their args to submit the local universe job. Then, the local universe job runs `snakemake_long.py long <user args>` to indicate to itself that it's time to submit the long-running Snakemake process instead of submitting another local universe job.

I'll add a comment for now to better explain the behavior, and if you don't like the name I can change it all together.